### ISSUE TITLE
Make Spotify interfaces return unwrapped paging objects

### ISSUE THREAD

**Issue by jzheng2017** [jzheng2017/spotify-web-api-wrapper#9]
**Is your feature request related to a problem? Please describe.**
Currently some api calls of the `SpotifyApi` interface returns an object wrapping a `Paging<T>` object. This is done because of the JSON the Spotify Web API returns.

For instance `PlaylistSimplifiedPaging` is just wrapping around `Paging<PlaylistSimplified>`. The end user would have to make an additional getter call to get the actual contents of the paging object. This is unnecessary because it is an internal detail of how the json is mapped.

The following functions need to be changed:
- [ ] [BrowseApi - getCategoryPlaylists](https://github.com/jzheng2017/spotify-web-api-wrapper/blob/41612ce3aa3f6783b6f1e7ab57c2b469bc80ad11/src/main/java/spotify/api/interfaces/BrowseApi.java#L16)
- [ ] [BrowseApi - getCategories](https://github.com/jzheng2017/spotify-web-api-wrapper/blob/41612ce3aa3f6783b6f1e7ab57c2b469bc80ad11/src/main/java/spotify/api/interfaces/BrowseApi.java#L18)
- [ ] [BrowseApi - getNewReleases](https://github.com/jzheng2017/spotify-web-api-wrapper/blob/41612ce3aa3f6783b6f1e7ab57c2b469bc80ad11/src/main/java/spotify/api/interfaces/BrowseApi.java#L22)

If there are anymore I've missed, please comment it on this issue.

**Describe the solution you'd like**
It would be better if the wrapped object contents got returned instead of the wrapper object itself.

So instead of returning the wrapped object, for instance `PlaylistSimplifiedPaging`, it would be better to returned the actual Paging object.

**Describe alternatives you've considered**
None

**Additional context**
The feature is actually already done once in `FollowApiRetrofit` class for the function `getFollowedArtists`. Take this function as reference when implementing the feature.

**Comment by Pratik-Nataraj516** [comment]
Hi, is this something I can work on?

**Comment by jzheng2017** [comment]
> Hi, is this something I can work on?

@Pratik-Nataraj516 Yes, of course. I'll assign this issue to you. When creating the branch, name the branch `feature/unwrap-paging-objects`. 

Also do not forget to read the [contribution guidelines](https://github.com/jzheng2017/spotify-web-api-wrapper/blob/main/CONTRIBUTING.md) before starting.

Good luck!

**Comment by jzheng2017** [comment]
@Pratik-Nataraj516 please confirm by 27-12-2020 that you're going to implement this feature. If there is no response by then, I'll have to remove you as assignee.

**Comment by Apoorvasaxena** [comment]
Looks interesting! Can I work on it?

**Comment by SumanohariR** [comment]
is this issue Still open? if yes, can I work on it?

**Comment by jzheng2017** [comment]
> is this issue Still open? if yes, can I work on it? I work as a backend dev and I think I can work on this.

Yeah, it's still open and you can work on this.

**Comment by SumanohariR** [comment]
Okay, Thank you for the confirmation.
I will start working on this

**Comment by jzheng2017** [comment]
> Okay, Thank you for the confirmation. I will start working on this

Any update on the progress of this issue?

**Comment by DanielVargaPublic** [comment]
Could you please take a look at the pull request resolving this issue? I've updated it as requested and I have a question about what version number to use.

**Comment by DanielVargaPublic** [comment]
Thank you, I updated the version numbers. Could you please create a release branch into which I could open a new pull request?

**Comment by jzheng2017** [comment]
> Thank you, I updated the version numbers. Could you please create a release branch into which I could open a new pull request?

I have created a 2.0.0 branch. Please use the old PR and target that branch.

**Comment by DanielVargaPublic** [comment]
> > Thank you, I updated the version numbers. Could you please create a release branch into which I could open a new pull request?
> 
> I have created a 2.0.0 branch. Please use the old PR and target that branch.

Thank you. I updated the base branch of the PR and re-requested the review.

**Comment by jzheng2017** [comment]
> > > Thank you, I updated the version numbers. Could you please create a release branch into which I could open a new pull request?
> > 
> > 
> > I have created a 2.0.0 branch. Please use the old PR and target that branch.
> 
> Thank you. I updated the base branch of the PR and re-requested the review.

Cool. Will take a look this week.

**Comment by DanielVargaPublic** [comment]
Hi! Could you please check the PR?

**Comment by jzheng2017** [comment]
https://github.com/jzheng2017/spotify-web-api-wrapper/pull/91 has been merged to [/2.0.0](https://github.com/jzheng2017/spotify-web-api-wrapper/tree/2.0.0) branch. This issue will be closed once the branch makes it to main.
